src/Product/Attributes/Color.php

```diff
@@ -15,6 +15,8 @@
  */
 class Color extends AbstractAttribute {
```

Comment:
This class should also implement WithMappingInterface, although if you follow the advice from the previous comment to create AbstractFieldAttribute then we can add the implementation in the abstract class to keep these simple. Same comment for the AgeGroup class.

Reply:
Since we are using interface methods in the `get_sources` function I think is better a Trait. 

I tried to setup `AbstractFieldAttribute` and `AbstractEnumAttribute` extending `AbstractAttribute` and implementing    `WithMappingInterface` but I got an error in execution because we are calling `self:get_id()` and `self::get_value_options()` from different interfaces. 

We can switch to Abstract maybe with some extra work but for now I've kept them with traits.